### ISSUE TITLE
Bugs that i found so far

### ISSUE THREAD

**Issue by 787992** [GaijinEntertainment/fmod_studio_warthunder_for_modders#15]
Hello there,
Creating this topic so you won't miss anything :)

Update on mg loop bugs: (loop never ends)
Not sure if it was fixed in future builds or not (everything tested on wip client)
This bug happens pretty often when you flying in plane over battlefield, almost 9 of 10 battles i could catch this bug, this is something related to sound distance, like passing by source of sound and 99% you will catch this bug.

Sound source (or emitters) are not physically attached to tank and not accurately following it's movement:
https://www.youtube.com/watch?v=dPa_s6Me1rM&feature=youtu.be
Sorry for reposting it again, just making sure devs will see it.

Recently, i discovered new interesting bug, before 1.97 update i could easily track cannon sounds by  killfeed, but now it's random (that's why i made supposition about randomly disappearing sounds after 1.97 update)
it's random because of how tank render system works or "invisible tanks." In other words, if tank in invisible state, you won't hear cannon shot, _however_ it ignores auto-cannons and mgs.

Distant cannon shots with vanilla quite tricky to hear, but this also happens when tank close by.
Sometimes you hear start of cannon shot and bam, sound suddenly cuts out, this is because in that very moment tank disappeared from player view. (it's also cuts out when someone destroyed right after shot)
https://youtu.be/je8TUYloIvg
Note, after shooting mg and auto cannon, my mate keeps shooting cannon, distance - around 1.1km
I also noticed if you can't see muzzleflash of cannon shot, you also won't hear cannon shot.

https://youtu.be/rJLKKsCHyx8
First thing, pretty much same as smoke laucher you can hear only one shot when you shoting multiple flares.
Second thing, this is rather question, in same video above you can hear ATGM engine noise attached to flares, is there any space so flares won't use ATGM noise for modding purposes?

**Comment by 787992** [comment]
TL;DR
Mg loop bug happens, when you flying in plane over battlefield.
Sound emitters for player's tank doesn't follow it's movement
Disappearing cannon shots related to non-rendered tanks (mg and autocannons are fine)
Flare laucher bugs out after one shot
Flares uses ATGM engine sounds

**Comment by Uchuujin01** [comment]
Hello! Great job, thank you for report! I'll check all listed bugs tomorrow (because of Russian holidays May 1-5).

**Comment by 787992** [comment]
In addition:
Jagdpanther and Coelian uses tiger engines w/ heavy tracks, i guess this is mistake?
There's more vehicles that using wrong engine events, i can create full list if you're interested.
Also discovered this:
![image](https://user-images.githubusercontent.com/63158562/82046756-30800080-96ba-11ea-929d-a52b405cfa5d.png)
Found these tabs for all tracks events, but all of them not working ingame, is that forgotten feature? Would be great to have dynamic sounds for tracks depending on surface.

**Comment by Uchuujin01** [comment]
> In addition:
> Jagdpanther and Coelian uses tiger engines w/ heavy tracks, i guess this is mistake?
> There's more vehicles that using wrong engine events, i can create full list if you're interested.
> Also discovered this:
> ![image](https://user-images.githubusercontent.com/63158562/82046756-30800080-96ba-11ea-929d-a52b405cfa5d.png)
> Found these tabs for all tracks events, but all of them not working ingame, is that forgotten feature? Would be great to have dynamic sounds for tracks depending on surface.

Hello! I know that there are some inappopriate engine sound references. I'm working on new engines sound events permanently, but it takes a lot of time.

Yes, not all this parameters working now, because of some maps circumstances. But we already working on a new dynamic terrain sound system, so all this current parameters will be deleted in future.

**Comment by 787992** [comment]
> Hello! I know that there are some inappopriate engine sound references. I'm working on new engines sound events permanently, but it takes a lot of time.

Oh hey! I appreciate your work on engines though, i think there's pretty much enough events for each engine type (except AMX ones and maybe leo1) 
I meant some tanks using wrong *existing* engine events, i.e. Coelian & jadgpanther uses Tiger while should be Panther engine instead.
Sorry for misunderstanding!

> But we already working on a new dynamic terrain sound system, so all this current parameters will be deleted in future.

Completely understandable, so we can expect some kind of same system? Would be lovely to hear different tracks sounds depending on surface!

Also, any update on bugs listed above? I can do more research if needed.

**Comment by Uchuujin01** [comment]
> Oh hey! I appreciate your work on engines though, i think there's pretty much enough events for each engine type (except AMX ones and maybe leo1)
> I meant some tanks using wrong _existing_ engine events, i.e. Coelian & jadgpanther uses Tiger while should be Panther engine instead.
> Sorry for misunderstanding!

Ah, understood, pardon me ) Yes, it would be great if you will send me a list of known wrong sound references. I do some historical research but sometimes miss out details.

> Completely understandable, so we can expect some kind of same system? Would be lovely to hear different tracks sounds depending on surface!

Yes, maybe even in 1.99 patch, but I can't say for sure now, we'll try )

> Also, any update on bugs listed above? I can do more research if needed.

No, I did not have enough time, but today I'll check it, thank you )

**Comment by 787992** [comment]
Alrighty, me and my mates decided to make full list of sounds (for every vehicle), it's nearly done. Are you guys accepting suggestions also? We can put in free colums of sounds what **could** be added. 


> No, I did not have enough time, but today I'll check it, thank you )
Regarding *when tank in invisible state you will not hear cannon shot* bug, i have now 100% confirmation: https://youtu.be/sLkYmgI5bI4
When enemy tank behind obstacle or wall or anything that hides it you won't hear cannon shot sound, although it does work for mg or aa guns.
Also pls watch other bugs that i listed :)

And last one: **The balance of the sound mix has been corrected towards lower frequencies.**
I noticed that this change affects mods, is it hardcoded or modders can change it in future fmod builds? It drastically changes sounds especially loud (which super quiet now) explosions or gunshots.

**Comment by Uchuujin01** [comment]
> Alrighty, me and my mates decided to make full list of sounds (for every vehicle), it's nearly done. Are you guys accepting suggestions also? We can put in free colums of sounds what could be added.

I think it could be useful, sure, very interesting!

> Regarding when tank in invisible state you will not hear cannon shot bug, i have now 100%

This bug was not confirmed. And, if it was so, then we'll be playing in silence, I think. On your video I  recognise far cannon shot.

> I noticed that this change affects mods, is it hardcoded or modders can change it in future fmod builds?

Yes, thanks for reminder. As I remember, now masterbank.bank loading from "sound" folder, not from "sound/mod". We'll give to modders ability to change mixer settings.

**Comment by 787992** [comment]
> We'll give to modders ability to change mixer settings.
Cheers, good to know

> This bug was not confirmed.
Look, in this case **hear** closely, my mate made shots to the ground to show that you can hear explosion but not the cannon itself, in second part he appeared and made more shots to show difference. You can make same steps as me, spawn in any map that has town, turn on screen share via Discord or other app and let your mate watch your shooting or vice versa, some maps has strict points where you can hide your tank from rendering, that's why town like Poland perfect choice.

Okay if that's not enough, here's the easiest way. Cannons can be heard from 2km, right? Spawn in same map like Poland and let your mate make some shots right from spawn, your mate/enemy should not move aswell as you to prevent rendering. However, if your mate will shoot from MG it will work flawlessly.
Please note, AA guns or rapid fire guns works also without any problems

> And, if it was so, then we'll be playing in silence, I think.
Well yes, but actually no. Tank disappears when nobody of teammates see enemies, basically you hearing shots from teammates but not enemies, which is also 50/50 situation, if teammate see enemy tank it will be rendered for everyone, however if enemy tank shooting from the hill or obstacle and no one spotted it, the sound will not be produced.

**Comment by Uchuujin01** [comment]
Ok, valuable note, thank you! It is very deep research, I'm impressed ) We'll research this problem, maybe this decision was made intentionaly.

**Comment by 787992** [comment]
> I think it could be useful, sure, very interesting!

[https://docs.google.com/spreadsheets/d/18zG_nibDyosUnKHMPZAhClDdzY0vcWypLPb-Gx_BQH4](url)
Voilà! Nothing really fancy here, this is mostly suggestions list. Added everything that more or less known + some of them easy to comapre with truck engines etc.
We also can do same with weapons (not sure what can be suggested here, probably only separate vehicles with open top turret) it will also take a while, if you interested here too.

**Comment by Uchuujin01** [comment]
Hello! Thank you, this is a great job, I'll apply your investigation results in the game. One question: what do you mean in suggestion "engine_m60" and engine_m41"? Does it means to make a unique m60 and m41 sounds?

**Comment by 787992** [comment]
> what do you mean in suggestion "engine_m60" and engine_m41"? Does it means to make a unique m60 and m41 sounds?

Yeah, as mentioned in notes starting from M48, US used pretty mch same engine contruction up to latest M60s, Merkavas etc.
![image](https://user-images.githubusercontent.com/63158562/83532434-eb036600-a4f6-11ea-864b-fee794763da2.png)

M41 used pretty much different engine type with distinct turbocharged sound

**Comment by Uchuujin01** [comment]
Ok, I'll keep it in mind, thank you! Great job )

**Comment by Uchuujin01** [comment]
BTW, regarding silent cannon shots when vehicle in invisible state. Our programmer confirmed that yes, now it works like that, and this is not a bug. I asked him to change this situation if it is possible. Thanks for the report!

**Comment by 787992** [comment]
> Ok, I'll keep it in mind, thank you! Great job )

o7, glad to contribute to something special!

> Our programmer confirmed that yes, now it works like that, and this is not a bug. I asked him to change this situation if it is possible

Wowie, didn't expect that, hope it will be changed, hearing almost half of battlefield sounds is a loss of immersion :(

**Comment by 787992** [comment]
@Uchuujin01 Could you please take a look at bug where sound emitters doesn't follow tank movement? With all bugs listed above i think this is major one. I'm noticing it more and more.

It's easy to check it, turn engine volume to minimum and rotate camera sideways, move tank forward or backwards, you will note how gear shift lever moving from back to forward (i.e is not standing in one place) it seems this happens for every emitter that attached to player's tank.
This pretty much started since 1.91 patch.
If you need more explanation let me know. 
https://youtu.be/dPa_s6Me1rM

**Comment by Uchuujin01** [comment]
> @Uchuujin01 Could you please take a look at bug where sound emitters doesn't follow tank movement? With all bugs listed above i think this is major one. I'm noticing it more and more.
> 
> It's easy to check it, turn engine volume to minimum and rotate camera sideways, move tank forward or backwards, you will note how gear shift lever moving from back to forward (i.e is not standing in one place) it seems this happens for every emitter that attached to player's tank.
> This pretty much started since 1.91 patch.
> If you need more explanation let me know.
> https://youtu.be/dPa_s6Me1rM

Hello! It’s a known issue, and it’s not a bug. Unfortunately, we unable to follow all emitted sounds due to cpu restrictions. We forced to switch play-and-forget mode on many game sounds, just to make game more stable.

**Comment by 787992** [comment]
> Hello! It’s a known issue, and it’s not a bug. Unfortunately, we unable to follow all emitted sounds due to cpu restrictions. We forced to switch play-and-forget mode on many game sounds, just to make game more stable.

Oh hey! Quite understandable, well, i wonder how it was made with FMOD Designer back then? Can't exactly remember same issues.

Alrighty, i think we sorted all bugs that i mentioned.
Last one more like question/feedback
> Flares uses ATGM engine sounds
Not a bug here, but any chance you could separate it from ATGM engine noises? The sound itself is decent for a single rocket, but if there many sounds (flares in this case) it makes pretty weird noises, especially when helicopter right above battlefield, kinda distracting.

**Comment by Uchuujin01** [comment]
> Oh hey! Quite understandable, well, i wonder how it was made with FMOD Designer back then? Can't exactly remember same issues.

It is not depends on Fmod, it depends on current game code stability of work. This situation changed constantly, game code changes constantly )

> Alrighty, i think we sorted all bugs that i mentioned.
> Last one more like question/feedback
> 
> > Flares uses ATGM engine sounds
> 
> Not a bug here, but any chance you could separate it from ATGM engine noises? The sound itself is decent for a single rocket, but if there many sounds (flares in this case) it makes pretty weird noises, especially when helicopter right above battlefield, kinda distracting.

Thank you, I will check it. Many different game objects shares the same game mechanics, sometimes sounds inherited by default )

**Comment by 787992** [comment]
@Uchuujin01 Noticed that most of sound mods stopped working (except vo mods)
What's changed?

**Comment by 787992** [comment]
> @Uchuujin01 Noticed that most of sound mods stopped working (except vo mods)
What's changed?

Never mind, fixed by deleting .masterbanks

So if you guys moving to 2.0 version, will be those "issues" fixed?
https://youtu.be/dPa_s6Me1rM
https://youtu.be/sLkYmgI5bI4

Oh, also i noticed that .masterbanks no more controlled by mod folder, is that going to be editable in next versions?

**Comment by Uchuujin01** [comment]
> Oh, also i noticed that .masterbanks no more controlled by mod folder, is that going to be editable in next versions?

Strange, It should work. Recently we added this feature. So, it worked some short time period, and then stopped? I'll check it.

> So if you guys moving to 2.0 version, will be those "issues" fixed?
> https://youtu.be/dPa_s6Me1rM
> https://youtu.be/sLkYmgI5bI4

The first one - definitely no, for now we decided to leave it like this. The second one - it is fixed, but we test it now, not sure it is work fine. We'll move this fix on production soon.

**Comment by Uchuujin01** [comment]
> Oh, also i noticed that .masterbanks no more controlled by mod folder, is that going to be editable in next versions?

Update on this issue. It works well now, I just checked on production client. Make sure you get the last update on your client. Now .masterbanks definitely controlled by mod folder.

**Comment by 787992** [comment]
> Update on this issue. It works well now, I just checked on production client. Make sure you get the last update on your client. Now .masterbanks definitely controlled by mod folder.

Ok, i'll check it

> The second one - it is fixed, but we test it now, not sure it is work fine. We'll move this fix on production soon.

Thank you very much for your time and effort!

**Comment by Uchuujin01** [comment]
> Thank you very much for your time and effort!

Thank you for your contribution and feedback!

**Comment by Uchuujin01** [comment]
> Ok, i'll check it

Hello! We found that something works wrong in mod sound banks loading. I hope today we'll find out what's wrong, and will fix it asap. I'll let you know when we'll figure out what's wrong exactly.

**Comment by Uchuujin01** [comment]
The problem in outdated version of this project. Some events now have different GUIDs. We loading events by GUID, not by string names. So, that's why some sounds not found. We can fix this problem for you only with actualisation of this project repo. I'll do this asap! The problem is - we just moved on new Fmod Studio version - 2.00.09 (on develop version of client), so it pointless, to publish this project in 1.10.17 version. But now on production version still 1.10.17.

**Comment by 787992** [comment]
> The problem in outdated version of this project. Some events now have different GUIDs. We loading events by GUID, not by string names. So, that's why some sounds not found. We can fix this problem for you only with actualisation of this project repo. I'll do this asap! The problem is - we just moved on new Fmod Studio version - 2.00.09 (on develop version of client), so it pointless, to publish this project in 1.10.17 version. But now on production version still 1.10.17.

Just wanted to write about it, probably that's why custom mixer setting doesn't work, only un-touched vanilla ones (from 1.97 version) which more than acceptable, since eq is fine here.
I wonder, how 2.0 version will improve sounds art/tech wise?

**Comment by Uchuujin01** [comment]
> I wonder, how 2.0 version will improve sounds art/tech wise?

In Fmod 2.0 Firelight team added a lot of cool features, (global parameters, modulation features, LFO, etc), looking forward to use it ) It's all described here, briefly: https://www.fmod.com/resources/documentation-studio?version=2.0&page=welcome-to-fmod-studio-new-in-200.html#whats-new-in-fmod-studio-20

**Comment by 787992** [comment]
> Sure we will. With new features it will be much easier now. Actually, in WT this feature not so important, this is not FPS, so we have very restricted cpu resourses on reverberation system.

Found another issue, not sure if that's intentional, it's related to tank hit sounds to armor.

I just noticed that all hit sounds for tank were reworked, but i never noticed them in battles!
We have 2 kind of hits:
1 for player tank and 2 for ai
ai hits intended (i guess) when you hear other tanks shooting to enemies/teammates, but that's not happens
You can hear ai hits only from your own shots.

**Comment by Uchuujin01** [comment]
> ai hits intended (i guess) when you hear other tanks shooting to enemies/teammates, but that's not happens

Hello! No, ai hits sound events is only for your shots on enemies. It works now how it's intended to be. But, yes, I want to make enemies and allies to hit each other with ai hit sound. This is in work plans on future. Thank you for your attention )

BTW, I followed your recomendations and just made a new engine sounds for M60 Patton. And separated tanks withM60 engines form tanks with Sherman engines, according to your engines sound list. So, it will be on production cirquit in two weeks, I think. Thanks for your assistance )

**Comment by 787992** [comment]
> Hello! No, ai hits sound events is only for your shots on enemies. It works now how it's intended to be. But, yes, I want to make enemies and allies to hit each other with ai hit sound. This is in work plans on future. Thank you for your attention )

That's what i thought at first glance, good to know.

> BTW, I followed your recomendations and just made a new engine sounds for M60 Patton. And separated tanks withM60 engines form tanks with Sherman engines, according to your engines sound list. So, it will be on production cirquit in two weeks, I think. Thanks for your assistance )

OKAY, This is epic ;) Glad to know that my little contribution did something special!

**Comment by Uchuujin01** [comment]
> OKAY, This is epic ;) Glad to know that my little contribution did something special!

Hello! Can you please give me a hand with one question? I found a very good recordings of a russian vehicle BRDM-2, ZMZ-41 engine, V8, 5,5 l, gasoline. This engine usually was used on military all-terrain vehicles. Where can we use it, on what vehicles?

**Comment by 787992** [comment]
> Hello! Can you please give me a hand with one question? I found a very good recordings of a russian vehicle BRDM-2, ZMZ-41 engine, V8, 5,5 l, gasoline. This engine usually was used on military all-terrain vehicles. Where can we use it, on what vehicles?

Heyo, i'm afraid that this is the only thing that using one engine for half-track GAZ-41 and BRDM family, however roots are coming from GAZ-11 which used by T-60, T-70, SU76, SU76m, ZIS-3, BM-8-24

**Comment by 787992** [comment]
> Hello! Can you please give me a hand with one question? I found a very good recordings of a russian vehicle BRDM-2, ZMZ-41 engine, V8, 5,5 l, gasoline. This engine usually was used on military all-terrain vehicles. Where can we use it, on what vehicles?

Heyo, it's nice to see you adding new engines, much thanks!
I have a question though, it's sounded before - will we see _in near future or in next patches_ different ai engines for tanks? Or still something with technical difficulties?

**Comment by Uchuujin01** [comment]
> > Hello! Can you please give me a hand with one question? I found a very good recordings of a russian vehicle BRDM-2, ZMZ-41 engine, V8, 5,5 l, gasoline. This engine usually was used on military all-terrain vehicles. Where can we use it, on what vehicles?
> 
> Heyo, i'm afraid that this is the only thing that using one engine for half-track GAZ-41 and BRDM family, however roots are coming from GAZ-11 which used by T-60, T-70, SU76, SU76m, ZIS-3, BM-8-24

Hello! Sorry, I missed out your answer somehow. Yes, that's what I decided too - T-60, T-70, SU76, SU76, and a couple of other vehicles.

> I have a question though, it's sounded before - will we see _in near future or in next patches_ different ai engines for tanks? Or still something with technical difficulties?

Definitely yes, but not so soon, there are will be much work to do.

**Comment by 787992** [comment]
> Definitely yes, but not so soon, there are will be much work to do.

Alrighty, good to know. 

Oh almost forgot, another thing that puzzles me, will helicopters receive ai sounds per heli model (not exactly individual)? Right now they are sharing same 3 events (piston, default and small one)

**Comment by Uchuujin01** [comment]
> Oh almost forgot, another thing that puzzles me, will helicopters receive ai sounds per heli model (not exactly individual)? Right now they are sharing same 3 events (piston, default and small one)

Mmm, I don't think it is necessary. As far as I know, helis sound almost the same from afar.

**Comment by 787992** [comment]
As far as I know, helis sound almost the same from afar.

It might seem at first glance but not. Because of different blades and rotor speed each of them has unique sounding especially Russian Mi-24 with it's agressive noise
https://www.youtube.com/watch?v=L6mSIE88CVU
US Helis tend to sound more smoother and less noisy
https://www.youtube.com/watch?v=eDvSrMnL-8M
https://www.youtube.com/watch?v=q0qKJ4Tev1c
No sure if ai ka-50 in game, but just in case from side it sounds different too
https://www.youtube.com/watch?v=uaRBeXBHdY4

**Comment by Uchuujin01** [comment]
> > Mmm, I don't think it is necessary. As far as I know, helis sound almost the same from afar.
> 
> It might seem at first glance but not. Because of different blades and rotor speed each of them has unique sounding especially Russian Mi-24 with it's aggressive noise
> 1- https://www.youtube.com/watch?v=L6mSIE88CVU
> 2- https://www.youtube.com/watch?v=peIpBZmztgs
> US Helis tend to sound more smoother and less noisy
> https://www.youtube.com/watch?v=eDvSrMnL-8M
> https://www.youtube.com/watch?v=q0qKJ4Tev1c
> No sure if ai ka-50 in game, but just in case from side it sounds different too
> https://www.youtube.com/watch?v=uaRBeXBHdY4

Sure, and it is implemented for player's helis, but for allies/enemies, I think it is ok. We'll see, it is not important task for now, anyway. Thank you for suggestion! I'll keep it.

**Comment by ChoukaiKaiNi** [comment]
I would also like to point out that with the Char B1 the 75mm and 47mm use the same sound.

**Comment by 787992** [comment]
> I would also like to point out that with the Char B1 the 75mm and 47mm use the same sound.

Good spot!

> Sure, and it is implemented for player's helis, but for allies/enemies, I think it is ok. We'll see, it is not important task for now, anyway. Thank you for suggestion! I'll keep it.

Found one issue related to explosions (not sure if it was inteded), previously we had parts_explosion when player/enemy tank respawns, now it shares with generic tank explosion (destroy).

**Comment by Uchuujin01** [comment]
> I would also like to point out that with the Char B1 the 75mm and 47mm use the same sound.

Thanks, I'll check it.

**Comment by Uchuujin01** [comment]
Hello! The project has been updated. Now it needs Fmod Studio 2.00.09.

**Comment by 787992** [comment]
> Hello! The project has been updated. Now it needs Fmod Studio 2.00.09.

Heyo, is it 2.0 compatible or i should transfer from 1.17 to 2.0?

**Comment by Uchuujin01** [comment]
> > Hello! The project has been updated. Now it needs Fmod Studio 2.00.09.
> 
> Heyo, is it 2.0 compatible or i should transfer from 1.17 to 2.0?

Hello! Well, older sound banks will work in game with fmod 2.00.09, but, due to fmod api changes, it will work not correcr. It is highly recommended to transfer your project to fmod 2.00.09 version.

**Comment by 787992** [comment]
> Hello! Well, older sound banks will work in game with fmod 2.00.09, but, due to fmod api changes, it will work not correcr. It is highly recommended to transfer your project to fmod 2.00.09 version.

Alrighty, switched to 2.0 version but this issue still exists (again, not sure if it was intentional)

> Found one issue related to explosions (not sure if it was inteded), previously we had parts_explosion when player/enemy tank respawns, now it shares with generic tank explosion (destroy).


On un-related note, do you have any plans expanding reload sounds? I really liking current approach of sounds, but that would be top notch if low-calibers tanks had their own reload sounds. Or simply separate 37/50/100/150mm reloads, especially ww2 tanks with autoloaders (which currently is very underrated such as Amx family)

**Comment by Uchuujin01** [comment]
> > Found one issue related to explosions (not sure if it was inteded), previously we had parts_explosion when player/enemy tank respawns, now it shares with generic tank explosion (destroy).

Hello! I'm not sure, why are you considered this an issue? parts_explosion launches when there are some parts exploded, and destroy event launches when vehicle destroyed - I don't see any issue.

> On un-related note, do you have any plans expanding reload sounds? I really liking current approach of sounds, but that would be top notch if low-calibers tanks had their own reload sounds. Or simply separate 37/50/100/150mm reloads, especially ww2 tanks with autoloaders (which currently is very underrated such as Amx family)

Sure, we will ) This work already in plans, thank you for your attention to details )

**Comment by 787992** [comment]
> Sure, we will ) This work already in plans, thank you for your attention to details )

💪

> parts_explosion launches when there are some parts exploded, and destroy event launches when vehicle destroyed - I don't see any issue.

The thing is parts_explosion is no more using anything, since Fmod designer it was used for any vehicle that respawns (that explosion when model disappears) Destroy was used for actual explosion after ammo rack for example or after j out. Now when tank or any any vehicle respawns from dead state it uses Destroy event instead of part_explosion.

**Comment by Uchuujin01** [comment]
> The thing is parts_explosion is no more using anything, since Fmod designer it was used for any vehicle that respawns (that explosion when model disappears) Destroy was used for actual explosion after ammo rack for example or after j out. Now when tank or any any vehicle respawns from dead state it uses Destroy event instead of part_explosion.

No, parts_explosion used now for destroyed parts of vehicle. And yes, destroy event uses as you wrote. I think that's how it should be.

**Comment by 787992** [comment]
> No, parts_explosion used now for destroyed parts of vehicle. And yes, destroy event uses as you wrote. I think that's how it should be.

Alrighty so this is not bug or issue.
Personally i really like sfx for explosion, especially this one, it's a great sound reward for killing tanks, but when it comes to flanking or when you playing alone - respawning vehicles nearby you with big loud explosion distracts more than plane overhead.
Parts_explosion had pretty subtle sound when vehicle respawns, that's why i thought it's a bug.
I recorded video just in case (first seconds cutted out, no idea why ;c .First seconds is actual explosion, second one when tank respawns - used exactly same event)
https://youtu.be/R7D4yyWFizc

Also found weird behaviour with engines, more like rpm thingy but not sure, noticed it reaally long time ago, couldn't find best example of it.
I tried to mod t-72 engine and found weird thing with rpm - offload, in editor you can edit seek speed, so offload will go off slower or faster for true life-like engine, but it seems game ignores seek speed values, i tried to test it with maximum seek speed, but it's all the same in game.
I thought something wrong with mod, so launched it with vanilla and same thing is here:
https://youtu.be/U7q9lYZu76s
It's basically with all ground vehicles, but t-72 the best example since it has distinct offload sounds.

**Comment by 787992** [comment]
Heyo, nice sound update, really liking those ATGM flybys. 
But only one thing i mentioned earlier, flares that uses helis and planes attached to ATGM rocket sound, and makes ultra weird noises.
https://www.youtube.com/watch?v=Yc19XYQzjqc

**Comment by 787992** [comment]
Nvm, apparently everything have been fixed
